### PR TITLE
Create `CreateEntityModal` component

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

This PR creates `CreateEntityModal` component. It's not used anywhere for now. But it'll be used soon for providing `properties` to blocks with good UX.

⚠️ There are few other small changes on this PR, explained at  _🔍 What does this change_ section below ⚠️ 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352032/1203701168177453/f) _(internal)_
- [Asana task](https://app.asana.com/0/1203543021352041/1203756383149343/f) _(internal)_

## 🔍 What does this change?

Here is the list of small fixes on this PR

- Move sidebar specific `Drawer` config from `MUI` theme to `sidebar.tsx`
- Create a reusable `use-draft-entity-subgraph` hook 
- Fix render issue for boolean values on `value-cell`
- Move `gap: 6.5px` from `EntityPageWrapper` to `EntityEditor` (to make entity editor styling more consistent)
- Comment out `PeersSection` on entity editor temporarily

## 🐾 Next steps

- Fix backend related issues to make the edit modal actually function with BP blocks.
- Moving `EntityEditor` and `EditEntityModal` into `/components` folder

## ❓ How to test this?

1.  Checkout the branch
1. Open any page and create open the block context bar
2. Click on "Edit Block" option
1.  Confirm that `CreateEntityModal` looks and works as expected.

## 📹 Demo

https://user-images.githubusercontent.com/39553853/213407380-335f7baf-5ef0-4226-ad8e-8f96c25934bd.mov

### CODE REVIEW

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx [hunk 1]
```diff
@@ -47,6 +62,11 @@ const BlockContextMenu: ForwardRefRenderFunction<
   { blockEntity, deleteBlock, openConfigMenu, popupState, canSwap },
   ref,
 ) => {
+  const { blockSubgraph, setBlockSubgraph } = useBlockContext();
+  const fetchBlockSubgraph = useFetchBlockSubgraph();
+
+  const [open, setOpen] = useState(false);
```
Comment (CiaranMn) [inline apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx:68]: This context menu has many items in it – I'd call this `entityEditorIsOpen` or something

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx [hunk 2]
```diff
@@ -98,6 +118,12 @@ const BlockContextMenu: ForwardRefRenderFunction<
           void navigator.clipboard.writeText(url.toString());
         },
       },
+      {
+        key: "edit-block",
+        title: "Edit Block",
+        icon: <FontAwesomeIcon icon={faGears} />,
```
Comment (CiaranMn) [inline apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx:124]: How about a pencil or something else? More edit-y

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx [hunk 3]
```diff
@@ -171,96 +197,125 @@ const BlockContextMenu: ForwardRefRenderFunction<
...
               (account) =>
                 account.entityId ===
                 blockEntity?.properties.entity.createdByAccountId,
             )?.preferredName
           } */}
-        </Typography>
+          </Typography>
+          {/** 
         {/** 
+          {/**
```
Comment (CiaranMn) [inline apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx:318]: ?

File: apps/hash-frontend/src/blocks/page/block-portals.tsx [hunk 4]
```diff
@@ -1,8 +1,10 @@
+// import { Subgraph, SubgraphRootTypes } from "@hashintel/hash-subgraph";
+import { Subgraph, SubgraphRootTypes } from "@blockprotocol/graph/.";
```
Comment (CiaranMn) [inline apps/hash-frontend/src/blocks/page/block-portals.tsx:2]:
```suggestion
import { Subgraph, SubgraphRootTypes } from "@blockprotocol/graph";
```

File: apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx [hunk 5]
```diff
@@ -393,10 +393,10 @@ const Page: NextPageWithLayout<PageProps> = ({
 
       <PageContextProvider pageEntityId={pageEntityId}>
         <Box
-          sx={({ zIndex, palette }) => ({
+          sx={({ palette }) => ({
             position: "sticky",
             top: 0,
-            zIndex: zIndex.appBar,
+            zIndex: 10,
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx:399]: We changed this from something connected to something else, to something independent – can we connect it back to something else? maybe `appBar + 1`? what is it fighting with?

**Review by CiaranMn — COMMENTED** [review]
A few minor comments on the code – looks good and works well otherwise (apart from the mentioned issue with block types being fixed at `Text`)

File: apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx (hunk 5 above)
Comment (yusufkinatas) [reply apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx]: Addressed here https://github.com/hashintel/hash/pull/1809/commits/916515b842b88016061425376e0fa7d75c5290fb. Removed a wrongly defined z-index from MUI theme options.

File: apps/hash-frontend/src/blocks/page/block-portals.tsx (hunk 4 above)
Comment (yusufkinatas) [reply apps/hash-frontend/src/blocks/page/block-portals.tsx]: Fixed here https://github.com/hashintel/hash/pull/1809/commits/916515b842b88016061425376e0fa7d75c5290fb

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx (hunk 3 above)
Comment (yusufkinatas) [reply apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx]: Fixed comment lines here https://github.com/hashintel/hash/pull/1809/commits/916515b842b88016061425376e0fa7d75c5290fb

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx (hunk 2 above)
Comment (yusufkinatas) [reply apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx]: Changed here https://github.com/hashintel/hash/pull/1809/commits/916515b842b88016061425376e0fa7d75c5290fb

File: apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx (hunk 1 above)
Comment (yusufkinatas) [reply apps/hash-frontend/src/blocks/page/block-context-menu/block-context-menu.tsx]: Changed here https://github.com/hashintel/hash/pull/1809/commits/916515b842b88016061425376e0fa7d75c5290fb

**Comment by yusufkinatas** [comment]
Thanks for the review! Addressed all of the comment above.

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**